sweep/sweeper.go

```diff
@@ -1861,20 +1874,139 @@ func (s *UtxoSweeper) handleBumpEvent(r *bumpResp) error {
 // NOTE: It is enough to check the txid because the sweeper will create
 // outpoints which solely belong to the internal LND wallet.
 func (s *UtxoSweeper) IsSweeperOutpoint(op wire.OutPoint) bool {
-	found, err := s.cfg.Store.IsOurTx(op.Hash)
-	// In case there is an error fetching the transaction details from the
-	// sweeper store we assume the outpoint is still used by the sweeper
-	// (worst case scenario).
-	//
-	// TODO(ziggie): Ensure that confirmed outpoints are deleted from the
-	// bucket.
-	if err != nil && !errors.Is(err, errNoTxHashesBucket) {
-		log.Errorf("failed to fetch info for outpoint(%v:%d) "+
-			"with: %v, we assume it is still in use by the sweeper",
-			op.Hash, op.Index, err)
+	return s.cfg.Store.IsOurTx(op.Hash)
+}
 
-		return true
+// markInputSwept marks the given input as swept by the tx. It will also notify
+// all the subscribers of this input.
+func (s *UtxoSweeper) markInputSwept(inp *SweeperInput, tx *wire.MsgTx) {
+	log.Debugf("Marking input as swept: %v from state=%v", inp.OutPoint(),
+		inp.state)
+
+	inp.state = Swept
+
+	// Signal result channels.
+	s.signalResult(inp, Result{
+		Tx: tx,
+	})
+
+	// Remove all other inputs in this exclusive group.
+	if inp.params.ExclusiveGroup != nil {
+		s.removeExclusiveGroup(*inp.params.ExclusiveGroup)
+	}
+}
+
+// handleUnknownSpendTx takes an input and its spending tx. If the spending tx
+// cannot be found in the sweeper store, the input will be marked as fatal,
+// otherwise it will be marked as swept.
+func (s *UtxoSweeper) handleUnknownSpendTx(inp *SweeperInput, tx *wire.MsgTx) {
+	op := inp.OutPoint()
+	txid := tx.TxHash()
+
+	isOurTx := s.cfg.Store.IsOurTx(txid)
+
+	// If this is our tx, it means it's a previous sweeping tx that got
+	// confirmed, which could happen when a restart happens during the
+	// sweeping process.
+	if isOurTx {
+		log.Debugf("Found our sweeping tx %v, marking input %v as "+
+			"swept", txid, op)
+
+		// We now use the spending tx to update the state of the inputs.
+		s.markInputSwept(inp, tx)
+
+		return
+	}
+
+	// Since the input is spent by others, we now mark it as fatal and won't
+	// be retried.
+	s.markInputFatal(inp, tx, ErrRemoteSpend)
+
+	log.Debugf("Removing descendant txns invalidated by (txid=%v): %v",
+		txid, lnutils.SpewLogClosure(tx))
+
+	// Construct a map of the inputs this transaction spends.
+	spentInputs := make(map[wire.OutPoint]struct{}, len(tx.TxIn))
+	for _, txIn := range tx.TxIn {
+		spentInputs[txIn.PreviousOutPoint] = struct{}{}
+	}
+
+	err := s.removeConflictSweepDescendants(spentInputs)
```

Comment:
CI is failing on an intermediate commit because `err` is redefined here.

The final code is fine though.